app/src/main/java/app/revanced/integrations/patches/components/HidePlayerFlyoutMenuFilter.java

```diff
@@ -0,0 +1,76 @@
+package app.revanced.integrations.patches.components;
+
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import app.revanced.integrations.settings.SettingsEnum;
+import app.revanced.integrations.shared.PlayerType;
+import app.revanced.integrations.utils.ReVancedUtils;
+
+public class HidePlayerFlyoutMenuFilter extends Filter {
+    private static String[] exceptions;
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    public HidePlayerFlyoutMenuFilter() {
+        exceptions = new String[]{
+                "comment", // Anything related to comment section.
+                "CellType|", // Comment filter chips on top of comment section.
+                "_sheet_", // Comment flyout panel for reporting and replying options.
+                "video_with_context" // Prevent video player lags sometimes when new video is started.
+        };
+
+        protobufBufferFilterGroups.addAll(
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_QUALITY_MENU,
+                        "yt_outline_gear"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_CAPTIONS_MENU,
+                        "yt_outline_closed_caption"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_LOOP_VIDEO_MENU,
+                        "yt_outline_arrow_repeat_1_"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_AMBIENT_MODE_MENU,
+                        "yt_outline_screen_light"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_REPORT_MENU,
+                        "yt_outline_flag"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_HELP_MENU,
+                        "yt_outline_question_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_MORE_INFO_MENU,
+                        "yt_outline_info_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_SPEED_MENU,
+                        "yt_outline_play_arrow_half_circle"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_AUDIO_TRACK_MENU,
+                        "yt_outline_person_radar"
+                ),
+                new ByteArrayAsStringFilterGroup(
+                        SettingsEnum.HIDE_WATCH_IN_VR_MENU,
+                        "yt_outline_vr"
+                )
+        );
+    }
+
+    @Override
+    boolean isFiltered(String path, String identifier, byte[] _protobufBufferArray) {
+        if (ReVancedUtils.containsAny(path, exceptions)) return false;
```

Comment:
Is it possible to turn this into a whitelist instead of a blacklist?

Reply:
elaborate?

Reply:
Instead of returning for a few cases, only allow filtering when it is certainly known, that it can be filtered

Reply:
Maybe if we revert this [commit](https://github.com/revanced/revanced-integrations/pull/416/commits/35a7022f67925810430f919a6ee1ac210eb23274).

Reply:
This will still not turn it into a whitelisted filter. You should check the pathBuilder string. Maybe you can do something along those lines:


```kotlin

if (path.contains(somePath))
  return super.filter(buffer)
```

This way you would prevent any kind of lag in the best possible way.